### PR TITLE
Address Lookup Field

### PR DESCRIPTION
**Pull request by macrael**

### CODE REVIEW

File: client/src/profile/AddressField.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import React, { PropTypes } from 'react';
+import AutoComplete from 'react-autocomplete';
+// import { reduxForm, Field } from 'redux-form';
+
+const MAPZEN_API_KEY = 'mapzen-Xt7d1Sz';
+const MAPZEN_AUTOCOMPLETE_URL = 'https://search.mapzen.com/v1/autocomplete';
+const SF_LAT = '37.7';
+const SF_LON = '-122.4';
+
+const styles = {
```
Comment (tenshiemi) [inline client/src/profile/AddressField.js]: Move the styles to custom.css

File: client/src/profile/AddressField.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+  highlightedItem: {
+    color: 'white',
+    background: 'hsl(200, 50%, 50%)',
+    padding: '2px 6px',
+    cursor: 'default',
+  },
+};
+
+function getURL(baseURL, getParams) {
```
Comment (tenshiemi) [inline client/src/profile/AddressField.js]: Move these functions in to the class itself

File: client/src/profile/AddressField.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
+import React, { PropTypes } from 'react';
+import AutoComplete from 'react-autocomplete';
+// import { reduxForm, Field } from 'redux-form';
```
Comment (tenshiemi) [inline client/src/profile/AddressField.js]: Remove since you're not using it

File: client/src/profile/AddressField.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        <label htmlFor="input-type-text">Add Address:</label>
+        <AutoComplete
+          value={this.state.value}
+          items={this.state.suggestions}
+          getItemValue={item => item.properties.label}
+          onSelect={(value, item) => {
+            this.setState({ value, suggestions: [item] }); // Probably unneccecary
+            this.props.saveAddress(item);
+          }}
+          onChange={(event, value) => {
```
Comment (tenshiemi) [inline client/src/profile/AddressField.js]: Pull out the function and make it a class method

File: client/src/profile/Profile.js [hunk 5]
```diff
@@ -1,8 +1,15 @@
 import React from 'react';
+import AddressField from './AddressField';
+
+function tmpStoreAddressObject(address) {
```
Comment (tenshiemi) [inline client/src/profile/Profile.js]: This should probably be passed down as a prop (and there is a current PR that adds a ProfileContainer.js component)

File: client/src/profile/AddressField.js (hunk 2 above)
Comment (macrael) [reply client/src/profile/AddressField.js]: They don't need access to this, though, so I will get an error if I do that, right?

Comment (tenshiemi) [reply client/src/profile/AddressField.js]: Just prefix the function name with the word 'static' e.g.
```
static submitUpdate(values) {
    console.log(values);
 }

File: client/styles/custom.scss [hunk 6]
```diff
@@ -16,7 +16,19 @@ $desktop: new-breakpoint(min-width 800px);
...
+  padding: 2px 6px;
+  cursor: default;
+}
+
+.highlightedAutocompleteItem {
+  color: white;
+  background: hsl(200, 50%, 50%);
+  padding: 2px 6px;
+  cursor: default;
+
```
Comment (tenshiemi) [inline client/styles/custom.scss]: Missing closing curly brace

**Pull request merged**